### DISCUSSION TITLE
API Capstone DA, Add data POST method vs Add manually to Data Base

### DISCUSSION THREAD

**Discussion by Talioll** [teamalgoritma/community#509] [Q&A] [answered]
Selamat siang, 

Sy dari batch Theia, saat mengerjakan capstone DA sy memilih web scrapping. Saat ini saya sedang mencoba mengerjakan capstone API Bikeshare untuk menambah repositories github. Mau tanya mengenai step adding new data untuk method post melalui flask, sebelum masuk ketahap ini di awal ada tahap dimana kita menambah data stations dan trips dari csv. Di tahap awal saya sudah input semua data trips tahun 2021 juga semua stations dengan metode loops di tahap awal, dengan contoh code seperti ini:

```python
def insert_into_trips(df, conn):
    for i in df.index:
        trip = tuple(df.iloc[i].values)
        query = f"""INSERT INTO trips values {trip}"""
        try:
            conn.execute(query)
        except:
            return 'Error'
        conn.commit()
    return 'OK'
```

kemudian di tahap selanjutnya ternyata ada tahap kita harus menambah data stations dan trips ke endpoint dengan metode post dengan flask dengan data dari csv juga (padahal semua data sudah sy input semua).

![api capstone](https://user-images.githubusercontent.com/112487401/211960020-7f68861e-69f5-40a6-a37a-9de3e4647dca.jpg)

Apakah memang seharusnya pada tahap awal tidak dimasukan semua data sehingga kita bisa menambah lagi data baru ke endpoint  yang belum diinput pada tahap awal? Saya agak bingung krn trips id dan station id sebagai unique key, dan khawatir kalau seharusnya di awal tahap tidak dimasukan semua data

**Comment by ranydwi** [comment, marked as answer]
Dear @Talioll  ,

Terima kasih atas pertanyaan yang disampaikan, pertanyaan Anda sudah kami terima. Kami akan mencoba membantu mencarikan referensi berkaitan dengan pertanyaan yang diberikan.

**Reply by ranydwi** [reply]
Dear @Talioll 

Untuk pertanyaan yang ibu tanyakan, sebetulnya ada beberapa hal yang bisa saya jawab yaitu:
- Penginputan data di awal tidak dilakukan secara langsung dengan looping ke databasenya melainkan melalui endpoint
- Untuk proses ini, silahkan Bu Talia mengikuti skeleton yang sudah disediakan oleh tim Algoritma saja.
- Sementara kekhawatiran Ibu di atas betul adanya, seharusnya tidak memasukan keseluruhan data di awal

Sekian. Semoga membantu

Regards.

**Reply by Talioll** [reply]
Dear mba @ranydwi,

Okee mba kalau begitu terima kasih atas konfirmasinya :)

**Reply by ranydwi** [reply]
Sama sama Bu @Talioll
